### DISCUSSION TITLE
false-positve warning `API resolved without sending a response` in image API response?

### DISCUSSION THREAD

**Discussion by garasubo** [vercel/next.js#40270] [Help] [answered]
I'm writing a code to returning image using Next.js API support, and running next dev server using `next dev` command, I got warning in the console `API resolved without sending a response for /api/hello, this may result in stalled requests.`.

According to the document, this waning happens when the api handler doesn't call methods to send data, like `res.send`. But actually, I called `res.send`.
Here is my code:
```
import type { NextApiRequest, NextApiResponse } from 'next'
import fs from "fs/promises";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<Object>
) {
  try {
    const data = await fs.readFile('./public/vercel.svg');
    res.setHeader('Content-Type', 'image/svg');
    res.status(200).send(data);
    console.log('end request');
  } catch (e) {
    res.status(404).end();
    console.log('error end request');
  }
}

``` 

And when I call this image using `next/image` component like this:
```
<Image src="/api/hello" alt="Vercel Logo" width={72} height={16} />
```

I got the warning `API resolved without sending a response for /api/hello, this may result in stalled requests.`.
Is that false-postive message, or am I something wrong?

I created this app using `npx create-next-app@latest --typescript`, and nothing changed except `pages/api/hello.ts` and the image src URL in `pages/index.tsx`.

Next.js version: 12.2.5
Node version: v16.13.1

Thanks.

**Comment by icyJoseph** [comment, marked as answer]
You need add this config export to the api route:

```js
export const config = {
  api: {
    externalResolver: true,
  },
}

```

I don't remember at the top of my head why using API routes with next/image triggers this but I'll go over my notes. I think I found it in a discussion from a while ago.

**Reply by garasubo** [reply]
Oh, nice. It worked! Thank you!

**Reply by sekomer** [reply]
this is the only working solution, thanks :')

**Reply by JoshW1WJosh** [reply]
This solution also worked for my project. Thanks for the help.

**Reply by almogzur** [reply]
OMG You Saved My Life THANK YOU

**Comment by Maha1295** [comment]
I have a problem with Next Js API.... when I add some JS file directly to the Pages folder the API works as expected but when I add some JS file to the api folder inside Pages folder it doesn't work as expected and I get a warning -> API resolved without sending a response for /api/hello1, this may result in stalled requests. 
`  const onContinueHandler = () => {
    router.push("/api/hello1");
  };`
in hello1.js I have the following code ->
`import React from "react";

const InputForm = () => {
  return <div>InputForm</div>;
};

export default InputForm;
`
And I tried adding your solution also and then the file looked like : 
`import React from "react";

const InputForm = () => {
  return <div>InputForm</div>;
};

export default InputForm;

export const config = {
  api: {
    externalResolver: true,
  },
};`
This time it didn't give me the error but the page kept on refreshing and did not navigated to expected page.
My folder structure looks like -> 
![filestructure](https://github.com/vercel/next.js/assets/127187824/b26ddfed-75d4-4962-9e27-0f15762b11eb)
but when I do router.push('/api/hello'); it works fine the code in hello.js is -> 
`// Next.js API route support: https://nextjs.org/docs/api-routes/introduction

export default function handler(req, res) {
  res.status(200).json({ name: 'John Doe' })
}`
and also when I do router.push('/dashboard') even that also works fine.... please help me out to understand what is happening internally
